go/internal/pkg/server/server.go

```diff
@@ -0,0 +1,50 @@
+package server
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+)
+
+type Config struct {
+	BaseEndpoint string
+}
+
+func NewHandler(cfg Config, httpClient *http.Client) Handler {
+	return Handler{cfg: cfg, httpClient: httpClient}
+}
+
+type Handler struct {
+	cfg        Config
+	httpClient *http.Client
+}
+
+func (h *Handler) ProxyHandle(w http.ResponseWriter, r *http.Request) {
+	url := h.cfg.BaseEndpoint + r.URL.Path
+	req, err := http.NewRequest(r.Method, url, r.Body)
+	if err != nil {
+		// TODO: Ignore error?
+		w.WriteHeader(http.StatusInternalServerError)
+		_, _ = fmt.Fprintf(w, "%v", err)
+		return
+	}
+
+	req.Header.Add("Content-Type", r.Header.Get("Content-Type"))
+	req.Header.Add("Content-Encoding", r.Header.Get("Content-Encoding"))
+	req.Header.Add("Accept", r.Header.Get("Accept"))
+	req.Header.Add("User-Agent", r.Header.Get("User-Agent"))
+	req.Header.Add("DD-API-KEY", r.Header.Get("DD-API-KEY"))
```

Comment:
I think duplicate all the headers of the incoming request would be more future proof.